### ISSUE TITLE
Express session not saving after successful authentication

### ISSUE THREAD

**Issue by gfcf14** [expressjs/session#679]
Hi. I am implementing a login system for my project. This project is divided in two, a server portion in `NodeJS`, and a client portion in `ReactJS`. Both of these are wrapped up in docker containers including a couple more containers for `mySQL` and `PHPMyAdmin`. Thus far, I've been able to connect to databases in the `mySQL` container and insert into a table for Users. Now, I'm trying to log in with a user, then save this user information if the login is successful, and return the session when asked. So I call the sign in get request as follows in the front-end:
```
export function signIn(table, userName, password) {
  return axios.get(`http://localhost:8000/signin`, {
    params: {
      table,
      userName,
      password,
    },
  }).then((response) => {
    if (response.data.length === 1) {
      return "success";
    }

    return response;
  });
}
```
Then in the server, I receive and work with the information like this:
```
const bcrypt = require('bcryptjs');
const bodyParser = require('body-parser');
const cors = require('cors');
const express = require('express');
const multer = require('multer');
const mysql = require('mysql');
const nodeMailer = require('nodemailer');
const session = require('express-session');
const smtpTransport = require('nodemailer-smtp-transport');

const app = express();
const upload = multer();

app.use(session({
    secret: 'secret',
    resave: true,
  saveUninitialized: true,
  cookie: {
    maxAge: 7 * 24 * 60 * 60 * 1000,
    secure: false,
  }
}));
app.use(cors());
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: false }));

const pool = mysql.createPool({
  host: process.env.MYSQL_HOST_IP,
  user: process.env.MYSQL_USER,
  password: process.env.MYSQL_PASSWORD,
  database: process.env.MYSQL_DATABASE,
});

app.get('/signin', (req, res) => {
  const { table, userName, password } = req.query;

  pool.query(`select * from ${table} where username = '${userName}'`, (err, results) => {
    if (err) {
      res.send(err);
    } else {
      if (bcrypt.compareSync(password, results[0].password)) {
        req.session.userId = results[0].id;
        req.session.name = results[0].name;
        req.session.email = results[0].email;
        req.session.sex = results[0].sex;
        req.session.img = results[0].img;
        req.session.userName = results[0].username;
        req.session.about = results[0].about;

        req.session.save(err => console.log(err));

        res.send(results);
      } else {
        res.send([]);
      }
    }
  });
});
```
Then I expect to call it with another request to get the information back and use to to modify a front end component's state like this (both of these requests are in the same file):
```
app.get('/loggeduser', (req, res) => {
  if (req.session.userId) {
    const {
      userId,
      name,
      email,
      sex,
      img,
      userName,
      about,
    } = req.session;

    const userInfo = {
      userId,
      name,
      email,
      sex,
      img,
      userName,
      about,
    };

    res.send(userInfo);
  } else {
    res.send({});
  }
});
```
and the component calls it like this:
```
export function getLoggedUser(setUserInfo) {
  axios.get(`http://localhost:8000/loggeduser`).then((response) => {
    setUserInfo(response.data);
  });
}
```
But the information never gets sent back, because `req.session.userId` is always undefined. I tried adding a console.log to output `req.session` and whenever I refresh the page (at which time the component calls `getLoggedUser`) the server image outputs `req.session` with a created time that is just a few seconds ago from the moment I refresh the page, meaning it gets created anew whenever I refresh. Is it that this is not saving properly because it's a get request and not a route? Please let me know if I may be missing something vital for this to work

**Comment by mlumbroso** [comment]
I am experiencing the same issue, the session gets refreshed with every new request made... Any lead so far ?

**Comment by sverlonom** [comment]
#685 May be it will help you @mlumbroso , @gfcf14

**Comment by gfcf14** [comment]
Thanks, @sverlonom , I tried that in several ways before; adding it first to `getLoggedUser` and then to `app.use(cors());`, then used both, but to no avail. I should probably close this, because I found a workaround, though I'm not sure it would fit your code, @mlumbroso . I used `js-cookie` ( https://www.npmjs.com/package/js-cookie ) to add the username from the user trying to log in. This creates a cookie to the local browser, and once it does I can read it later. I only add the username because that's all I really need to verify a login (I think!)
